packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -978,12 +978,12 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
       0.0,
       constraints.maxWidth - (
         _boxSize(icon).width
-        + contentPadding.left
+        + (prefixIcon !=null ? 0: contentPadding.left)
```

Comment:
I think prefixIcon ends up on the leading edge (per RTL vs LTR) and suffixIcon on the trailing edge. So I think you'd have to write something like:
```dart
final EdgeInsets horizontalContentPadding = Directionality.of(context) == TextDirection.ltr ? contentPadding : contentPadding.flipped;
```
And then use horizontalContentPadding instead of contentPadding.